app/views/AddExistingChannelView.js

```diff
@@ -79,13 +80,15 @@ class AddExistingChannelView extends React.Component {
 					Q.experimentalSortBy('room_updated_at', Q.desc)
 				)
 				.fetch();
-			const filteredChannels = channels.filter(async(channel) => {
+
+			const filteredChannels = await asyncFilter(channels, async(channel) => {
 				const permissions = await RocketChat.hasPermission([addTeamChannelPermission], channel.rid);
-				if (!permissions[0]) {
-					return;
+				if (!permissions[0] || channel.prid) {
```

Comment:
Can we move this `channel.prid` up, so `hasPermission` is not executed when it's a discussion?